### ISSUE TITLE
Incorrect Syntax Highlight with std::filesystem

### ISSUE THREAD

**Issue by TH3CHARLie** [microsoft/vscode-cpptools#4415]
**Type: LanguageService**

**Describe the bug**
- OS and Version: MacOS Mojave 10.14.6
- VS Code Version: 1.39.0
- C/C++ Extension Version: 0.26.0-insiders3
- Other extensions you installed (and if the issue persists after disabling them): None
- A clear and concise description of what the bug is.

After aliasing the std::filesystem namespace, using the alias to call constructor leads to incorrect syntax highlighting.

**To Reproduce**

1. Use code sample below:
```c++
#include <filesystem>
#include <iostream>
namespace fs = std::filesystem;
const std::string FOLDER_PATH = "./";
int main() {
  for (auto& entry : fs::recursive_directory_iterator(FOLDER_PATH)) {
    std::cout << entry.path() << std::endl;
  }
  for (auto& entry : fs::directory_iterator(FOLDER_PATH)) {
    std::cout << entry.path() << std::endl;
  }
}
```
2. See incorrect highlighting in line 6 and 9,  namely ```fs::r``` and ```fs::d```.
3. My ```c_cpp_properties.json```:
```json
{
    "configurations": [
        {
            "name": "Mac",
            "includePath": [
                "${workspaceFolder}/**"
            ],
            "defines": [],
            "macFrameworkPath": [
                "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks"
            ],
            "compilerPath": "/usr/bin/clang",
            "cStandard": "c11",
            "cppStandard": "c++17",
            "intelliSenseMode": "clang-x64"
        }
    ],
    "version": 4
}
```

**Expected behavior**

The letter 'r'/'d' on line 6 and 9 should be consistent with rest of the class "recursive_directory_iterator"/"directory_iterator"

**Screenshots**

https://photos.app.goo.gl/JVocYgkznkPEhmM57

**Additional context**

**Comment by Colengms** [comment]
Hi @TH3CHARLie .  Thanks for letting us know.  :)

**Comment by daandemeyer** [comment]
I have similar issues in comments:

![image](https://user-images.githubusercontent.com/9395011/66700163-fc8cc900-eced-11e9-9ce2-362b55a8f338.png)

https://github.com/DaanDeMeyer/reproc/blob/2de7a05360d39d495942c48730c804623a5398f5/reproc/include/reproc/reproc.h#L205

I'm not actually using the C/C++ extension however. Do I have to make an issue in the main repository or does this repository also handle syntax highlighting?

**Comment by Colengms** [comment]
Hi @DaanDeMeyer .  If not using the C/C++ Extension, I would suggest reporting the issue in the main VS Code repo: https://github.com/microsoft/vscode

**Comment by Colengms** [comment]
This seems to be related to the length of the namespace alias:

![image](https://user-images.githubusercontent.com/49173979/101095839-6bbd2080-3573-11eb-814d-8af315484844.png)

It also doesn't appear to repro when not used as part of a range-based for loop.

**Comment by Colengms** [comment]
Although there is a different symptom, I'm seeing a related issue in VS (which we share IntelliSense code with).  In VS, hover operations seem to be returning the wrong results within range-based for loops (which we also repro, though slightly differently - only with the first character in the token).  It's possible that these incorrect positions are affecting our colorization ranges.  That issue is here: https://developercommunity.visualstudio.com/content/problem/1277449/c-intellisense-hover-over-symbol-in-a-range-based.html

**Comment by Colengms** [comment]
The linked VS issue still exists, but the original repro here appears to have been fixed (in 0.29.0).  Closing this as fixed.
